File: FitToFit/UI/fittofit_mobile/lib/models/vjezbe.dart
import 'package:fittofit_mobile/models/vjezbeTreninzi.dart';
import 'package:json_annotation/json_annotation.dart';

part 'vjezbe.g.dart';

@JsonSerializable()
class Vjezbe {
  int vjezbaId;
  String naziv;
  String? opis;
  String? slika;
  List<VjezbeTreninzi>? treninziVjezbes = [];

  Vjezbe(
      {required this.vjezbaId,
      required this.naziv,
      this.opis,
      this.slika,
      this.treninziVjezbes});

  factory Vjezbe.fromJson(Map<String, dynamic> json) => _$VjezbeFromJson(json);

  Map<String, dynamic> toJson() => _$VjezbeToJson(this);
}

File: FitToFit/FitToFit.Services/Database/Vjezbe.cs
using System;
using System.Collections.Generic;

namespace FitToFit.Services.Database;

public partial class Vjezbe
{
    public int VjezbaId { get; set; }

    public string Naziv { get; set; } = null!;

    public string? Opis { get; set; }

    public byte[]? Slika { get; set; }

    public virtual ICollection<TreninziVjezbe> TreninziVjezbes { get; set; } = new List<TreninziVjezbe>();
}
